### ISSUE TITLE
Default route for multiple networks

### ISSUE THREAD

**Issue by DCsunset** [containers/podman#16233]
**Is this a BUG REPORT or FEATURE REQUEST? (leave only one on its own line)**

/kind feature

**Description**

Currently, when a container is connected to multiple networks, multiple conflicting default routes will be added (shown by `ip route`), which makes it hard to control which network will be used by default.

It would be easier to control the routing if podman supports an option to specify which network to be used by default. Then only one default route should be created.

**Comment by mheon** [comment]
We are adding an option for per-network route metrics, which should allow what you want. This will only be supported by the `netavark` backend, of note.

**Comment by Cydox** [comment]
This creates an issue especially if a container is connected to one internal network (created with `--internal`) and one "normal" network. In that case you get two default routes and the container can't reach the internet if the route for the internal network happens to appear as the first route in `ip route` (order appears to be random).

Example:

```
$ podman network create --internal int
$ podman network create ext
$ podman run --rm -it --network int --network ext docker.io/alpine:3.16 ping example.com
```

Running the last line of the above example repeatedly shows that the ping works about 50% of the time.

This same example works using docker. With docker you have to create the container first with only one network and then connect the other network. Docker example:

```
$ docker network create --internal int
$ docker network create ext
$ docker create --rm -it --network ext --name test docker.io/alpine:3.16
$ docker network connect int test
$ docker start test
$ docker exec -it test ping example.com
```

Looking in more detail at the behavior of docker it looks like they change out the default route of the container on the fly as you connect and disconnect the `ext` network while the container is running. So they appear to not do this by precedence of which network was connected first but the external network always takes precedence over the internal one.

I'm not sure if this should be considered a feature request or a bug. What behavior is currently documented for these cases and what should the behavior be?

Should I open a new issue for this or keep this in here?

Note: This is with the netavark backend, haven't tested with CNI.

**Comment by Luap99** [comment]
The internal route is a bug and already fixed in netavark, https://github.com/containers/podman/issues/16256

**Comment by Cydox** [comment]
Great! Was using podman 4.2.1 and netavark 1.1.0.

**Comment by Luap99** [comment]
@mheon Is all your stuff merged? I think this should work now.

**Comment by mheon** [comment]
Concur. Going to close, will be fixed in Podman 4.4, probably out in January
